rocAL/include/pipeline/tensor.h

```diff
@@ -65,14 +65,21 @@ void allocate_host_or_pinned_mem(void** ptr, size_t size, RocalMemType mem_type)
 struct Roi {
     unsigned* get_ptr() { return _roi_ptr.get(); }
     Roi2DCords* get_2D_roi() {
-        if (_roi_no_of_dims != 2)
-            THROW("ROI has more than 2 dimensions. Cannot return Roi2DCords")
-        return reinterpret_cast<Roi2DCords*>(_roi_ptr.get());
+        auto out_index = sizeof(Roi2DCords);
```

Comment:
Where is the condition check, this is applicable only for selected layouts right?